malariagen_data/af1.py

```diff
@@ -0,0 +1,747 @@
+                        the MalariaGEN website</a> or contact data@malariagen.net.
+                    </td></tr>
+                </thead>
+                <tbody>
+                    <tr>
+                        <th style="text-align: left">
+                            Storage URL
+                        </th>
+                        <td>{self._url}</td>
+                    </tr>
+                    <tr>
+                        <th style="text-align: left">
+                            Data releases available
+                        </th>
+                        <td>{', '.join(self.releases)}</td>
+                    </tr>
+                    <tr>
+                        <th style="text-align: left">
+                            Results cache
+                        </th>
+                        <td>{self._results_cache}</td>
+                    </tr>
+                    <tr>
+                        <th style="text-align: left">
+                            Site filters analysis
+                        </th>
+                        <td>{self._site_filters_analysis}</td>
+                    </tr>
+                    <tr>
+                        <th style="text-align: left">
+                            Software version
+                        </th>
+                        <td>malariagen_data {malariagen_data.__version__}</td>
+                    </tr>
+                    <tr>
+                        <th style="text-align: left">
+                            Client location
+                        </th>
+                        <td>{self._client_location}</td>
+                    </tr>
+                </tbody>
+            </table>
+        """
+        return html
+
+    # TODO: generalise (species, cohorts) so we can abstract to parent class
+    def _sample_metadata(self, *, sample_set):
+        df = self._read_general_metadata(sample_set=sample_set)
+        # TODO: when cohorts are available
+        # df_cohorts = self._read_cohort_metadata(sample_set=sample_set)
+        # df = df.merge(df_cohorts, on="sample_id", sort=False)
+        return df
+
+    def _transcript_to_gene_name(self, transcript):
+        df_geneset = self.geneset().set_index("ID")
+        rec_transcript = df_geneset.loc[transcript]
+        parent = rec_transcript["Parent"]
+        rec_parent = df_geneset.loc[parent]
+
+        # TODO: tailor to Af
+        # manual overrides
+        # if parent == "AGAP004707":
+        #     parent_name = "Vgsc/para"
+        # else:
+        #     parent_name = rec_parent["Name"]
+        parent_name = rec_parent["Name"]
+
+        return parent_name
+
+    def _site_mask_ids(self):
+        if self._site_filters_analysis == "dt_20200416":
+            return ["funestus"]
+        elif self._site_filters_analysis == "sc_20220908":
+            return ["funestus"]
+        else:
+            raise ValueError
+
+    def _snp_variants_dataset(self, *, contig, inline_array, chunks):
```

Comment:
Could this method be moved to the base class?

Reply:
Yes, that shouldn't be there, thanks. Will fix.